db/migrate/20230928211022_add_directory_to_globus_destination.rb

```diff
@@ -0,0 +1,17 @@
+class AddDirectoryToGlobusDestination < ActiveRecord::Migration[7.0]
+  def up
+    add_column :globus_destinations, :directory, :string
+
+    # populate any missing directory values
+    GlobusDestination.all.each do |dest|
+      dest.directory = dest.created_at.strftime('%Y-%m-%d-%H-%M-%S-%L')
+      dest.save
+    end
+
+    change_column_null :globus_destinations, :directory, false
+  end
+
+  def down
+    drop_column :globus_destinations, :directory
+  end
+end
```

Comment:
Because serializing and parsing dates with micro-seconds was a bit tedious I ended up adding the Directory field to the GlobusDeposit model. This makes it easier to set and get it without needing to infer it from the created_at timestamp. It also means we can easily change the default directory name in the future if we want.